packages/calypso-e2e/src/lib/blocks/jetpack-extended/instagram.ts

```diff
@@ -0,0 +1,61 @@
+import { Frame } from 'playwright';
+import { BlockFlow, EditorContext, PublishedPostContext } from '..';
+
+interface ConfigurationData {
+	embedUrl: string;
+	expectedPostText: string;
+}
+
+const blockParentSelector = '[aria-label="Block: Embed"]:has-text("Instagram URL")';
+const selectors = {
+	embedUrlInput: `${ blockParentSelector } input`,
+	embedButton: `${ blockParentSelector } button:has-text("Embed")`,
+	editorInstagramIframe: `iframe[title="Embedded content from instagram.com"]`,
+	publishedInstagramIframe: `iframe.instagram-media`,
+};
+
+/**
+ * Class representing the flow of using an Instagram block in the editor
+ */
+export class InstagramBlockFlow implements BlockFlow {
+	private configurationData: ConfigurationData;
+
+	/**
+	 * Constructs an instance of this block flow with data to be used when configuring and validating the block.
+	 *
+	 * @param {ConfigurationData} configurationData data with which to configure and validate the block
+	 */
+	constructor( configurationData: ConfigurationData ) {
+		this.configurationData = configurationData;
+	}
+
+	blockSidebarName = 'Instagram';
```

Comment:
Are we shifting to the `blockSidebarName` syntax from existing `blockName`?

Reply:
Oh i had just done that as a way to be explicit about where that name was used. I can switch it back! Or are you suggesting that we update the parameter name in `addBlock` too?

Reply:
It was more of a reference to the existing block classes like `PricingTable`, which still uses the `blockName` convention.

Reply:
Ah yeah, I intentionally for this first PR didn't want to touch (apart from satisfying transpiler errors) any of the existing block tests / POMs. I figured we could rectify those with this in a separate PR! But in that case, I will probably update the member name then

Reply:
That's fine by me, let's update it at a later date.